partiql-eval/src/eval/eval_expr_wrapper.rs

```diff
@@ -50,6 +50,12 @@ impl TypeSatisfier for Static {
                             | (Static::DateTime, Value::DateTime(_))
                     )
                 }
+                DatumValueRef::Lower(lower) => {
+                    // TODO this basically clones just to throw the value away after type-check; fix that
+                    let lowered = lower.lower();
+                    let lowered = lowered.expect("lower");
+                    self.satisfies(&lowered)
```

Comment:
Should we be keeping this extra clone?

Reply:
We'll have to fix at a later time; more advanced type machinery is needed first.

FWIW, in all of all the conformance tests and additional test added to exercise this boxed ion, these lines aren't hit:  https://app.codecov.io/gh/partiql/partiql-lang-rust/pull/536?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=partiql

I'll try to devise a test to cover it to add to #545. But it is at least rather uncommon.

Reply:
I actually don't think it's possible to get to this branch during normal evaluation in `partiql-eval`. Prior to type-checking, the value will have already been lowered by https://github.com/partiql/partiql-lang-rust/pull/540/files/7f79f58f251f2450c7f6febd97c5772fc4ce99f4#diff-b3a236452a3b5f704e7f2d2fa80983f726caed63e4dade21bf90b48c4b6546a6R353-R356

Reply:
I see. Mind if we delete these lines then if we can confirm this branch won't get hit?